Apply-WindowsUpdateArtifactToVMs.ps1

```diff
@@ -0,0 +1,69 @@
+﻿param
+(
+    [Parameter(Mandatory=$false, HelpMessage="Configuration File, see example in directory")]
+    [ValidateNotNullOrEmpty()]
+    [string] $ConfigFile = "config.csv",
+
+    [Parameter(HelpMessage="String containing comma delimitated list of patterns. The script will apply updates to just the VMs matching one of the patterns.")]
+    [string] $ImagePattern = "",
+
+    [Parameter(Mandatory=$false, HelpMessage="Shutdown the VMs when done?")]
+    [ValidateNotNullOrEmpty()]
+    [boolean] $shutdownVMs = $false
+
+)
+
+$ErrorActionPreference = "Stop"
+
+. "./Utils.ps1"
+
+Import-AzDtlModule
+
+# Read in the config file
+$config = Import-Csv $ConfigFile
+
+# Foreach lab, get all the VMs
+$config | ForEach-Object {
+    Write-Host "----------------------------------------------------------" -ForegroundColor Green
+    Write-Host "Applying Windows Updates to all VMs in DevTestLab: $($_.DevTestLabName) matching pattern: '$($ImagePattern)'" -ForegroundColor Green
+    $vms = Get-AzDtlVm -Lab @{Name = $_.DevTestLabName; ResourceGroupName = $_.ResourceGroupName}
+    
+    # Trim down the list of VMs based on the image pattern passed in...
+    if ($ImagePattern) {
```

Comment:
This is similar to the Select-VmSettings commandlet in utils.ps1 script - but we're operating on the list of VMs instead of list of SourceImageInfos.  I didn't generalize the utils.ps1 function because I'm hoping to publish a version of this script more generally for customers.